src/ape/api/accounts.py

```diff
@@ -180,6 +183,11 @@ def deploy(self, contract: "ContractContainer", *args, **kwargs) -> "ContractIns
         logger.success(f"Contract '{contract_name}' deployed to: {styled_address}")
         instance = ContractInstance.from_receipt(receipt, contract_type)
         self.chain_manager.contracts.cache_deployment(instance)
+
+        if publish:
+            address = AddressType(HexAddress(HexStr(receipt.contract_address)))
```

Comment:
It is an optional string: https://github.com/ApeWorX/ape/blob/main/src/ape/api/transactions.py#L148
Probably should handle the optional.

Reply:
It should be optional, but probably `Optional[AddressType]`

It should throw in the case it is `None`, because there might be a scenario where it fails to deploy

Reply:
Can this change be made on this PR? I think it only breaks `mypy` since `AddressType` is a str alias